globus_cli/parsing/command_state.py

```diff
@@ -36,17 +39,35 @@ def is_verbose(self):
 def format_option(f):
     def callback(ctx, param, value):
         state = ctx.ensure_object(CommandState)
-        # need to do an OR check here because this is invoked with value=None
-        # everywhere that the `-F`/`--format` option is omitted (each level of
-        # the command tree)
-        state.output_format = (value or state.output_format).lower()
+        # only set a new format if there is no stored jmespath expression
+        # otherwise, it must remain JSON
+        if state.jmespath_expr is None:
+            # need to do an OR check here because this is invoked with
+            # value=None everywhere that the `-F`/`--format` option is
+            # omitted (each level of the command tree)
+            state.output_format = (value or state.output_format).lower()
         return state.output_format
 
-    return click.option(
+    def jmespath_callback(ctx, param, value):
+        if value is None:
+            return
+        state = ctx.ensure_object(CommandState)
+        state.output_format = 'json'
+        state.jmespath_expr = jmespath.compile(value)
+        return state.jmespath_expr
+
+    f = click.option(
         '-F', '--format',
         type=CaseInsensitiveChoice([JSON_FORMAT, TEXT_FORMAT]),
         help='Output format for stdout. Defaults to text',
         expose_value=False, callback=callback)(f)
+    f = click.option(
+        "--jmespath",
```

Comment:
I don't really mind `--query`, but I worried that it might have bad optics on commands like `globus endpoint search` where "query" is a meaningful verb. Is that not a big deal or not as confusing as I worry?

The other thing on my mind with its naming is general skittishness about having a short option for this. JMESPath queries aren't for the faint hearted -- a malformed expression will blow up, sometimes rather opaquely. Should it have a short opt?